### PR TITLE
Updating npm symlink when installing from source

### PR DESCRIPTION
**Pull request by sjlu**
Should fix issue #90 where the symlink gets overridden

### CODE REVIEW

**Comment by BarthV** [comment]
please rebase into a single commit.

:ok: 

what about this @guilhem ?

File: recipes/nodejs_from_binary.rb [hunk 1]
```diff
@@ -35,13 +35,15 @@
...
-  binaries = ['bin/iojs', 'bin/node', 'bin/npm']
+  binaries = ['bin/iojs', 'bin/node']
 else
   filename = "node-v#{node['nodejs']['version']}-linux-#{arch}.tar.gz"
   archive_name = 'nodejs-binary'
-  binaries = ['bin/node', 'bin/npm']
+  binaries = ['bin/node']
 end
 
+binaries.push('bin/npm') if node['nodejs']['npm']['install_method'] != 'source'
```
Comment (guilhem) [inline recipes/nodejs_from_binary.rb]: can you check `== 'embedded'` instead?

**Comment by BarthV** [comment]
After review this PR, it appears that this is not a correct way to handle nodejs (from bin) + npm (from anything else that source).

In fact your are not checking if we are calling npm recipe effectively (with something like `node.recipes.include?(recipe_name)` ...), but only testing the value of an attributes (this attribute can be used or not...).

I'm thinking about a elegant way to handle nodejs + npm in the same runlist to avoid double symlinks. But this is really not the best solution to me.

File: recipes/nodejs_from_binary.rb (hunk 1 above)
Comment (BarthV) [inline recipes/nodejs_from_binary.rb]: :+1:

**Comment by sjlu** [comment]
@BarthV 

I can look into it having it check for if the recipe has ran or not (forewarning: I'm not an expert in Chef).

This was the only way I found to having a custom version of npm installed easily without having the symlink overwritten.

* Installing npm through the npm chef script - but this just gets blown away after Chef runs again cause it updates the symlink and then reinstalls npm again :-/
* npm from source - again, this gets blown away after another run then reinstalled and built from source

I'm all ears though and would love to help make a get a solution going.

**Comment by guilhem** [comment]
@sjlu You just have to use `embedded`, squash your commits and I will merge it :)

**Comment by sjlu** [comment]
@guilhem fixed & squashed

**Comment by guilhem** [comment]
Good to me :)

**Comment by BarthV** [comment]
:+1:

**Pull request merged**